### PR TITLE
feat: timeline plugin rendering optimization

### PR DESCRIPTION
**Pull request by lelenei**
## Description

Closes # mf-412

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/plugin-infra/src/types.ts [hunk 1]
```diff
@@ -244,6 +245,8 @@ export namespace Plugin.SNSAdaptor {
         ProfileSliders?: ProfileSlider[]
         /** This UI will be rendered as tabs on the profile page */
         ProfileTabs?: ProfileTab[]
+        /** This UI will be rendered as plugin wrapper page */
+        wrapperEntry?: WrapperEntry
```
Comment (Jack-Works) [inline packages/plugin-infra/src/types.ts]: Rename this field to `PluginWrapperProps`

File: packages/plugin-infra/src/types.ts [hunk 2]
```diff
@@ -336,6 +339,11 @@ export namespace Plugin.SNSAdaptor {
         category?: 'dapp' | 'other'
     }
 
+    export interface PluginWrapperProps {
+        icon?: React.ReactNode
+        style?: CSSProperties
```
Comment (hanszhoou) [inline packages/plugin-infra/src/types.ts]: Seems `background` will always be the only property of `style`.

```ts
export interface PluginWrapperProps {
    icon?: React.ReactNode
    backgroundGradient?: string
}
```

File: packages/mask/src/web3/UI/EthereumChainBoundary.tsx [hunk 3]
```diff
@@ -239,7 +254,7 @@ export function EthereumChainBoundary(props: EthereumChainBoundaryProps) {
     return renderBox(
         <>
             {!noSwitchNetworkTip ? (
-                <Typography color="textPrimary">
+                <Typography color="#FF3545">
```
Comment (hanszhoou) [inline packages/mask/src/web3/UI/EthereumChainBoundary.tsx]: Create a variable for `#FF3545` in `packages/theme/src/CSSVariables`

File: packages/mask/src/web3/UI/EthereumChainBoundary.tsx [hunk 4]
```diff
@@ -185,7 +193,7 @@ export function EthereumChainBoundary(props: EthereumChainBoundaryProps) {
 
     if (!isAllowed)
         return renderBox(
-            <Typography color="textPrimary">
+            <Typography color="#07101B">
```
Comment (hanszhoou) [inline packages/mask/src/web3/UI/EthereumChainBoundary.tsx]: Create a variable for `#07101B` in `packages/theme/src/CSSVariables`

File: packages/mask/src/components/InjectedComponents/DisabledPluginSuggestion.tsx [hunk 5]
```diff
@@ -55,21 +59,62 @@ export function PossiblePluginSuggestionUI(props: { plugins: Plugin.DeferredDefi
     const { t } = useI18N()
     const t2 = usePluginI18NField()
```
Comment (hanszhoou) [inline packages/mask/src/components/InjectedComponents/DisabledPluginSuggestion.tsx]: rename `t2` to `i18n`

File: packages/mask/src/components/InjectedComponents/DisabledPluginSuggestion.tsx [hunk 6]
```diff
@@ -55,21 +59,62 @@ export function PossiblePluginSuggestionUI(props: { plugins: Plugin.DeferredDefi
...
     return (
         <>
             {plugins.map((x) => (
                 <MaskPostExtraInfoWrapper
+                    open
                     key={x.ID}
-                    title={t('plugin_not_enabled', { plugin: t2(x.ID, x.name) })}
+                    title={t2(x.ID, x.name)}
+                    publisher={
+                        x.publisher ? <PluginI18NFieldRender pluginID={x.ID} field={x.publisher.name} /> : undefined
```
Comment (hanszhoou) [inline packages/mask/src/components/InjectedComponents/DisabledPluginSuggestion.tsx:75]: Replace `<PluginI18NFieldRender  />` with usage of `usePluginI18NField()`?

File: packages/mask/src/plugins/Collectible/SNSAdaptor/ActionBar.tsx [hunk 7]
```diff
@@ -10,13 +10,15 @@ import { CheckoutDialog } from './CheckoutDialog'
...
 const useStyles = makeStyles()((theme) => {
     return {
-        root: {
-            marginLeft: theme.spacing(-0.5),
-            marginRight: theme.spacing(-0.5),
-        },
+        root: {},
         button: {
             flex: 1,
             margin: `0 ${theme.spacing(0.5)}`,
```
Comment (hanszhoou) [inline packages/mask/src/plugins/Collectible/SNSAdaptor/ActionBar.tsx]:
```suggestion
            margin: theme.spacing(0, 0.5),
```

File: packages/mask/src/components/InjectedComponents/DisabledPluginSuggestion.tsx [hunk 8]
```diff
@@ -52,25 +54,63 @@ export function PossiblePluginSuggestionPostInspector() {
...
+        borderRadius: 8,
+    },
+}))
+interface RectangleProps extends HTMLProps<HTMLDivElement> {}
+
+export function Rectangle(props: RectangleProps) {
+    const { classes } = useRectangleStyles()
+    return (
+        <div {...props}>
+            <div className={classes.rectangle} style={{ width: 103 }} />
```
Comment (guanbinrui) [inline packages/mask/src/components/InjectedComponents/DisabledPluginSuggestion.tsx]: Try this https://mui.com/material-ui/react-skeleton/.

File: packages/mask/src/components/InjectedComponents/DisabledPluginSuggestion.tsx [hunk 9]
```diff
@@ -71,6 +72,7 @@ export function PossiblePluginSuggestionUI(props: { plugins: Plugin.DeferredDefi
                         x.publisher ? <PluginI18NFieldRender pluginID={x.ID} field={x.publisher.name} /> : undefined
                     }
                     publisherLink={x.publisher?.link}
+                    wrapperProps={_plugins.filter((y) => y.ID === x.ID)?.[0]?.wrapperProps}
```
Comment (septs) [inline packages/mask/src/components/InjectedComponents/DisabledPluginSuggestion.tsx]:
```suggestion
                    wrapperProps={_plugins.find((y) => y.ID === x.ID)?.wrapperProps}
```

File: packages/icons/plugins/MarketsClaim.tsx [hunk 10]
```diff
@@ -16,3 +16,109 @@ export const MarketsClaimIcon: typeof SvgIcon = createIcon(
     </>,
     '0 0 36 36',
 )
+
+const svg = (
+    <svg width="46" height="46" viewBox="0 0 46 46" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (septs) [inline packages/icons/plugins/MarketsClaim.tsx]: plz use https://aka.pw/svgomg reduce svg image

File: packages/mask/src/plugins/FindTruman/base.ts [hunk 11]
```diff
@@ -9,7 +9,7 @@ export const base: Plugin.Shared.Definition = {
     description: {
         fallback: 'A plugin for https://findtruman.io/',
     },
-    publisher: { name: { fallback: 'FindTruman' }, link: 'https://findtruman.io/' },
+    publisher: { name: { fallback: 'Mask Network' }, link: 'https://mask.io/' },
```
Comment (Jack-Works) [inline packages/mask/src/plugins/FindTruman/base.ts]: ?

File: packages/mask/src/plugins/PoolTogether/base.tsx [hunk 12]
```diff
@@ -3,9 +3,9 @@ import { POOLTOGETHER_PLUGIN_ID, URL_PATTERN } from './constants'
 
 export const base: Plugin.Shared.Definition = {
     ID: POOLTOGETHER_PLUGIN_ID,
-    name: { fallback: 'PoolTogether' },
+    name: { fallback: 'Pooltogether' },
     description: { fallback: 'PoolTogether is a protocol for no-loss prize games on the Ethereum blockchain' },
-    publisher: { name: { fallback: 'iRhonin' }, link: 'https://github.com/iRhonin' },
+    publisher: { name: { fallback: 'Mask Network' }, link: 'https://mask.io/' },
```
Comment (Jack-Works) [inline packages/mask/src/plugins/PoolTogether/base.tsx]: ?

File: packages/plugins/CyberConnect/src/base.tsx [hunk 13]
```diff
@@ -9,7 +9,7 @@ export const base: Plugin.Shared.Definition = {
     description: {
         fallback: 'A plugin for https://cyberconnect.me/',
     },
-    publisher: { name: { fallback: 'CyberConnect' }, link: 'https://github.com/cyberconnecthq' },
+    publisher: { name: { fallback: 'Mask Network' }, link: 'https://mask.io/' },
```
Comment (Jack-Works) [inline packages/plugins/CyberConnect/src/base.tsx]: ?

File: packages/plugins/FileService/src/SNSAdaptor/index.tsx [hunk 14]
```diff
@@ -36,7 +36,7 @@ const definition: Plugin.SNSAdaptor.Definition = {
     ApplicationEntries: [
         (() => {
             const icon = <FileServiceIcon />
-            const name = { i18nKey: '__plugin_name', fallback: 'File Service' }
+            const name = { fallback: 'File Service' }
```
Comment (Jack-Works) [inline packages/plugins/FileService/src/SNSAdaptor/index.tsx]: please revert this change.

File: packages/plugins/FileService/src/base.ts [hunk 15]
```diff
@@ -3,7 +3,7 @@ import { languages } from './locales/languages'
 
 export const base: Plugin.Shared.Definition = {
     ID: PluginId.FileService,
-    name: { fallback: 'File Service', i18nKey: '__display_name' },
+    name: { fallback: 'Files Service' },
```
Comment (Jack-Works) [inline packages/plugins/FileService/src/base.ts:6]: Please revert this change.

File: packages/plugins/FileService/src/base.ts [hunk 16]
```diff
@@ -4,7 +4,7 @@ import { META_KEY_1, META_KEY_2 } from './constants'
...
```
Comment (Jack-Works) [inline packages/plugins/FileService/src/base.ts]: Please revert this change.

File: packages/plugins/FileService/src/SNSAdaptor/index.tsx (hunk 14 above)
Comment (Jack-Works) [reply packages/plugins/FileService/src/SNSAdaptor/index.tsx]: Please revert this change.

File: packages/mask/src/components/InjectedComponents/DisabledPluginSuggestion.tsx (hunk 8 above)
Comment (Jack-Works) [reply packages/mask/src/components/InjectedComponents/DisabledPluginSuggestion.tsx]: @lelenei

File: packages/mask/src/web3/UI/EthereumWalletConnectedBoundary.tsx [hunk 17]
```diff
@@ -21,10 +32,12 @@ export interface EthereumWalletConnectedBoundaryProps
     children?: React.ReactNode
     hideRiskWarningConfirmed?: boolean
     ActionButtonProps?: ActionButtonProps
+    startIcon?: React.ReactNode
+    hiddenPersonaVerified?: boolean
 }
 
 export function EthereumWalletConnectedBoundary(props: EthereumWalletConnectedBoundaryProps) {
-    const { children = null, offChain = false, hideRiskWarningConfirmed = false } = props
+    const { children = null, offChain = false, hideRiskWarningConfirmed = false, hiddenPersonaVerified = false } = props
```
Comment (Jack-Works) [inline packages/mask/src/web3/UI/EthereumWalletConnectedBoundary.tsx]: This props is unused.

File: packages/mask/src/web3/UI/EthereumChainBoundary.tsx [hunk 18]
```diff
@@ -85,7 +114,7 @@ export function EthereumChainBoundary(props: EthereumChainBoundaryProps) {
         WalletMessages.events.connectWalletDialogUpdated,
```
Comment (Jack-Works) [inline packages/mask/src/web3/UI/EthereumChainBoundary.tsx]: This is unused.

File: packages/mask/src/web3/UI/EthereumChainBoundary.tsx [hunk 19]
```diff
@@ -138,37 +167,54 @@ export function EthereumChainBoundary(props: EthereumChainBoundaryProps) {
             pluginIDSettings.value = NetworkPluginID.PLUGIN_EVM
```
Comment (Jack-Works) [inline packages/mask/src/web3/UI/EthereumChainBoundary.tsx]: Unused.

File: packages/mask/src/plugins/dHEDGE/base.tsx [hunk 20]
```diff
@@ -5,7 +5,7 @@ export const base: Plugin.Shared.Definition = {
     ID: DHEDGE_PLUGIN_ID,
     name: { fallback: 'dHEDGE' },
     description: { fallback: 'Decentralized hedge funds on Ethereum.' },
-    publisher: { name: { fallback: 'iRhonin' }, link: 'https://github.com/iRhonin' },
+    publisher: { name: { fallback: 'Mask Network' }, link: 'https://mask.io/' },
```
Comment (Jack-Works) [inline packages/mask/src/plugins/dHEDGE/base.tsx]: ?

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacket/index.tsx [hunk 21]
```diff
@@ -51,7 +50,7 @@ export function RedPacket(props: RedPacketProps) {
         value: availability,
         computed: availabilityComputed,
         retry: revalidateAvailability,
-    } = useAvailabilityComputed(account, payload)
+    } = useAvailabilityComputed(account ?? payload.contract_address, payload)
 
     const { NATIVE_TOKEN_ADDRESS } = useTokenConstants()
```
Comment (Jack-Works) [inline packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacket/index.tsx:53]: tokenDetailed is unused. please remove it.

**Review by Jack-Works — CHANGES_REQUESTED** [review]

File: packages/mask/src/plugins/dHEDGE/base.tsx (hunk 20 above)
Comment (lelenei) [reply packages/mask/src/plugins/dHEDGE/base.tsx]: image

File: packages/mask/src/components/InjectedComponents/DisabledPluginSuggestion.tsx [hunk 22]
```diff
@@ -52,25 +54,69 @@ export function PossiblePluginSuggestionPostInspector() {
...
+                                fontSize: 14,
+                                fontWeight: 700,
+                                paddingTop: 1,
+                                paddingBottom: 1.125,
+                                lineHeight: 1.5,
+                                '&:hover': {
+                                    backgroundColor: MaskColorVar.buttonPluginBackground,
+                                },
+                            }}>
+                            Enable plugins
```
Comment (guanbinrui) [inline packages/mask/src/components/InjectedComponents/DisabledPluginSuggestion.tsx]: i18n

File: cspell.json [hunk 23]
```diff
@@ -277,6 +277,7 @@
         "openocean",
         "pausable",
         "pkts",
+        "pluginid",
```
Comment (nuanyang233) [inline cspell.json]: It looks like this pr is not using this word, please delete it

File: packages/mask/src/plugins/PoolTogether/base.tsx (hunk 12 above)
Comment (nuanyang233) [reply packages/mask/src/plugins/PoolTogether/base.tsx]: Please revert this change, we can't change publisher info!

File: packages/mask/src/components/InjectedComponents/DisabledPluginSuggestion.tsx [hunk 24]
```diff
@@ -52,25 +54,69 @@ export function PossiblePluginSuggestionPostInspector() {
...
+                        </Button>
```
Comment (nuanyang233) [inline packages/mask/src/components/InjectedComponents/DisabledPluginSuggestion.tsx]: Why need these hard code styles, And please add i18n.

File: packages/mask/src/plugins/EVM/hooks/useAsset.ts [hunk 25]
```diff
@@ -17,14 +18,15 @@ export function useAsset(address: string, tokenId: string, provider: NonFungible
     const { WNATIVE_ADDRESS } = useTokenConstants()
 
     return useAsyncRetry(async () => {
-        const asset = await EVM_RPC.getAsset({ address, tokenId, chainId, provider })
+        const asset = await EVM_RPC.getAsset({ address, tokenId, chainId: ChainId.Mainnet, provider })
```
Comment (guanbinrui) [inline packages/mask/src/plugins/EVM/hooks/useAsset.ts]: Are you sure to use a fixed chain id? It's a breaking change.

File: packages/mask/src/plugins/Furucombo/UI/PoolView.tsx [hunk 26]
```diff
@@ -87,6 +88,8 @@ export function PoolView(props: PoolProps) {
     const { classes } = useStyles()
     const { t } = useI18N()
     const { category, chainId, address, name, protocol, liquidity, apy, angels } = props.investable
+    const account = useAccount()
+    const _chainId = useChainId()
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Furucombo/UI/PoolView.tsx]: `useChainIdMatched`

File: packages/mask/src/plugins/Gitcoin/base.ts [hunk 27]
```diff
@@ -12,13 +10,6 @@ export const base: Plugin.Shared.Definition = {
         architecture: { app: true, web: true },
         networks: { type: 'opt-out', networks: {} },
         target: 'stable',
-        web3: {
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Gitcoin/base.ts:15]: Why remove them?

File: packages/mask/src/plugins/ITO/SNSAdaptor/hooks/useAvailability.ts [hunk 28]
```diff
@@ -1,16 +1,18 @@
 import { useAsyncRetry } from 'react-use'
-import { useAccount, useITOConstants, useChainId, isSameAddress } from '@masknet/web3-shared-evm'
+import { useAccount, useITOConstants, useChainId, isSameAddress, ChainId } from '@masknet/web3-shared-evm'
 import { checkAvailability } from '../../Worker/apis/checkAvailability'
 
-export function useAvailability(id: string, contract_address: string) {
+export function useAvailability(id: string, contract_address: string, from?: string, chainId?: ChainId) {
     const account = useAccount()
-    const chainId = useChainId()
```
Comment (guanbinrui) [inline packages/mask/src/plugins/ITO/SNSAdaptor/hooks/useAvailability.ts:7]: Always keep the same naming style. The camelCase is used across this project.

File: packages/mask/src/plugins/ITO/SNSAdaptor/hooks/useAvailabilityComputed.ts [hunk 29]
```diff
@@ -7,9 +7,10 @@ import { useQualification } from './useQualification'
 import { ITO_CONTRACT_BASE_TIMESTAMP } from '../../constants'
 
 export function useAvailabilityComputed(payload: JSON_PayloadInMask | JSON_PayloadFromChain) {
-    const chainId = useChainId()
+    const chain_id = useChainId()
+    const chainId = payload.chain_id ?? chain_id
```
Comment (guanbinrui) [inline packages/mask/src/plugins/ITO/SNSAdaptor/hooks/useAvailabilityComputed.ts]: The reason the payload uses snake-case is it is an external data format. In project code, we prefer camelCase.

File: packages/mask/src/plugins/MaskPluginWrapper.tsx [hunk 30]
```diff
@@ -14,104 +11,105 @@ import {
...
+            <Typography variant="body1" fontSize={14} fontWeight="500" color={MaskColorVar.textPluginColor}>
                 {publisher}
             </Typography>
         )
         return (
-            <Box>
-                <Typography variant="h6" fontSize="1.1rem" fontWeight="400" color={MaskColorVar.textSecondary}>
-                    Provided by
+            <Box className={classes.provider}>
+                <Typography sx={{ marginRight: 0.5 }} variant="body1" fontSize={14} fontWeight="400" color="#767F8D">
```
Comment (guanbinrui) [inline packages/mask/src/plugins/MaskPluginWrapper.tsx]: Why not replace `#767F8D` into a mask color?

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/hooks/useAvailability.ts [hunk 31]
```diff
@@ -1,16 +1,15 @@
...
     return useAsyncRetry(async () => {
-        if (!id || redpacketChainId !== currentChainId) return null
+        if (!id) return null
         if (!redPacketContract) return null
         return redPacketContract.methods.check_availability(id).call({
             // check availability is ok w/o account
             from,
         })
-    }, [id, from, redpacketChainId, currentChainId, redPacketContract])
+    }, [id, from, redpacketChainId, redPacketContract])
```
Comment (guanbinrui) [inline packages/mask/src/plugins/RedPacket/SNSAdaptor/hooks/useAvailability.ts:14]: @zhouhanseng Take a look at this.

File: packages/mask/src/plugins/dHEDGE/base.tsx (hunk 20 above)
Comment (guanbinrui) [reply packages/mask/src/plugins/dHEDGE/base.tsx]: You don't need to follow the Figma about the publisher. It's customizable.

File: packages/web3-shared/evm/hooks/useERC20TokenDetailed.ts [hunk 32]
```diff
@@ -8,21 +8,18 @@ import type { ERC20Bytes32 } from '@masknet/web3-contracts/types/ERC20Bytes32'
...
     const currentChainId = useChainId()
     const chainId = targetChainId ?? currentChainId
     const erc20TokenContract = useERC20TokenContract(address, chainId)
     const erc20TokenBytes32Contract = useERC20TokenBytes32Contract(address, chainId)
-    const tokenType = useEthereumTokenType(address)
 
     return useAsyncRetry(async () => {
         if (!address) return
         if (!EthereumAddress.isValid(address)) return
-        if (tokenType !== EthereumTokenType.ERC20) return
```
Comment (guanbinrui) [inline packages/web3-shared/evm/hooks/useERC20TokenDetailed.ts]: Why add this change?

File: packages/mask/src/plugins/Gitcoin/base.ts (hunk 27 above)
Comment (lelenei) [reply packages/mask/src/plugins/Gitcoin/base.ts:15]: required figma

File: packages/mask/src/plugins/ITO/SNSAdaptor/hooks/useAvailability.ts [hunk 33]
```diff
@@ -1,16 +1,17 @@
...
+    const _chainId = useChainId()
 
     const { ITO_CONTRACT_ADDRESS } = useITOConstants()
     const isV1 = isSameAddress(contract_address ?? '', ITO_CONTRACT_ADDRESS)
 
     return useAsyncRetry(async () => {
         if (!id || !contract_address) return null
-        return checkAvailability(id, account, contract_address, chainId, isV1)
+        return checkAvailability(id, fromAccount, contract_address, chainId ?? _chainId, isV1)
     }, [id, account])
```
Comment (guanbinrui) [inline packages/mask/src/plugins/ITO/SNSAdaptor/hooks/useAvailability.ts]: You lost something here.

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/hooks/useRedPacketContract.ts [hunk 34]
```diff
@@ -6,20 +6,20 @@ import type { HappyRedPacketV1 } from '@masknet/web3-contracts/types/HappyRedPac
...
         HAPPY_RED_PACKET_ADDRESS_V2: addressV2,
         HAPPY_RED_PACKET_ADDRESS_V3: addressV3,
         HAPPY_RED_PACKET_ADDRESS_V4: addressV4,
-    } = useRedPacketConstants()
-    const v1 = useContract<HappyRedPacketV1>(addressV1, HappyRedPacketV1ABI as AbiItem[])
-    const v2 = useContract<HappyRedPacketV2>(addressV2, HappyRedPacketV2ABI as AbiItem[])
-    const v3 = useContract<HappyRedPacketV3>(addressV3, HappyRedPacketV3ABI as AbiItem[])
-    const v4 = useContract<HappyRedPacketV4>(addressV4, HappyRedPacketV4ABI as AbiItem[])
+    } = useRedPacketConstants(chainId)
+    const v1 = useContract<HappyRedPacketV1>(addressV1, HappyRedPacketV1ABI as AbiItem[], chainId)
```
Comment (guanbinrui) [inline packages/mask/src/plugins/RedPacket/SNSAdaptor/hooks/useRedPacketContract.ts:19]: In the coming `refactor/migrate-evm`, the `chainId` goes the first parameter of all `useXXContract`. You could do the same thing in this PR too.

```suggestion
    const v1 = useContract<HappyRedPacketV1>(chainId, addressV1, HappyRedPacketV1ABI as AbiItem[])
```

File: packages/mask/src/plugins/Snapshot/SNSAdaptor/VotingCard.tsx [hunk 35]
```diff
@@ -69,29 +73,30 @@ export function VotingCard() {
...
+                            variant="outlined">
+                            {choiceText}
+                        </Button>
+                    ))}
+
+                    <Button
+                        className={classes.button}
+                        disabled={choice === 0 || !account || !power}
+                        onClick={() => setOpen(true)}>
+                        {Boolean(power) && Boolean(account) ? t('plugin_snapshot_vote') : t('plugin_snapshot_no_power')}
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Snapshot/SNSAdaptor/VotingCard.tsx]:
```suggestion
                        {power && account ? t('plugin_snapshot_vote') : t('plugin_snapshot_no_power')}
```

**Review by guanbinrui — APPROVED** [review]
